ivi-layermanagement-api/ilmControl/src/ilm_control_wayland_platform.c

```diff
@@ -2136,11 +2138,20 @@ static void screenshot_done(void *data, struct ivi_screenshot *ivi_screenshot,
 
     munmap(buffer, size);
 
-    if (save_as_bitmap(filename, (const char *)image_buffer,
-                       image_size, width, height, bytes_per_pixel * 8) == 0) {
-        ctx_scrshot->result = ILM_SUCCESS;
+    if ((filename_ext = strstr(filename, ".png")) && (strlen(filename_ext) == 4)) {
+        if (save_as_png(filename, (const char *)image_buffer,
+                              width, height, bytes_per_pixel * 8, has_alpha) == 0) {
+            ctx_scrshot->result = ILM_SUCCESS;
+        } else {
+            fprintf(stderr, "failed to write screenshot as png file: %m\n");
+        }
     } else {
-        fprintf(stderr, "failed to write screenshot file: %m\n");
+        if (save_as_bitmap(filename, (const char *)image_buffer,
+                        image_size, width, height, bytes_per_pixel * 8) == 0) {
+            ctx_scrshot->result = ILM_SUCCESS;
+        } else {
+            fprintf(stderr, "failed to write screenshot file: %m\n");
```

Comment:
Looks like `as bmp` is missing in this message. In case of png you are printing `"failed to write screenshot as png file: %m\n"`